### DISCUSSION TITLE
Iniciar uma variável

### DISCUSSION THREAD

**Discussion by gthemadd** [Flutterando/forum#133] [Perguntas e Respostas]
Boa tarde! Sou nova no flutter e estou usando o tamanho/índice da lista para adicionar o valor as minhas variáveis , porém recebo o seguinte erro "The instance member can't be accessed in an initializer"

Já tentei adicionar o late e colocar a função dentro da build, o que parou o erro, mas agora ela vem vazia

 Alguém tem alguma idéia do que possa ser? Segue meu código abaixo

![IMG-20230313-WA0013.jpg](https://user-images.githubusercontent.com/94639880/224781045-ebdf43db-26c0-4c6b-85e7-2fe8e3a08e2e.jpg)

![IMG-20230313-WA0014.jpg](https://user-images.githubusercontent.com/94639880/224781044-15906f83-2df9-471a-bb6c-76076aa81fd8.jpg)

**Comment by EduardoM-GO** [comment]
Bom dia @gthemadd, pelo os trechos de código acima, creio que tu consiga corrigir assim:

` getListImage(service_orders_id, realm_id_question, number) async {
      list_Image[number] = await ImagemController.getListImage(
          service_orders_id, realm_id_question);
      setState(() {});
      return list_Image[number];
    }
`

**Reply by 30103010** [reply]
tenta usar o wait quando tiver que esperar resultado de funcoes asyncronas

**Reply by ghost** [reply]
Pelo que vi no seu código e com o destaque da sintaxe parece e talvez que esteja utilizando o JetBrains, não sei se você sabe mais geralmente a JetBrains(IDE), tem um "analyze" para corrigir esses problemas de variáveis. Eu tentaria clicar no botão direito do mouse e verificar se não aparece uma lista suspensa semelhante para incluir essa variável.

Espero ajudar inicialmente, já que pensei que utiliza o JetBrains(IDE) para o flutter. Se isso ajudar, informe. Caso não, vou ver como ajudar de outra forma.
